generic/ras.py

```diff
@@ -40,6 +40,14 @@ def setUp(self):
         architecture = os.uname()[4]
         if "ppc" not in architecture:
             self.skip("supported only on Power platform")
+        if 'PowerNV' in process.system_output('grep ^platform /proc/cpuinfo'):
+            nv_flag = False
+            for cmd in ['lsmcode', 'lsprop', 'nvram', 'ppc64_cpu']:
+                if cmd in self.name.name:
```

Comment:
Anyway, if you need granularity here, I'd instead suggest you to use the skip decorators. Example:
```python
from avocado import Test
from avocado import skipUnless

IS_POWER_NV = 'PowerNV' in open('/proc/cpuinfo', 'r').read()

class MyTest(Test):
    def setUp(self):
        pass

    @skipUnless(IS_POWER_NV, "Skipping test in non-PowerNV platform")
    def test3_lsmcode(self):
        pass

    @skipUnless(IS_POWER_NV, "Skipping test in non-PowerNV platform")
    def test5_lsprop(self):
        pass
```